### PR TITLE
Add component name and Write node to instance

### PR DESCRIPTION
**Pull request by mkolar**
Adds component name from ftrack aware write node. If its just plain write node, it does nothing.

Changed default 'None' component to 'main' to comply with ftrack defaults.

Adding write node to the instance data, so it can be potentially queried in the validators.

### CODE REVIEW

File: pyblish_deadline/plugins/select_deadline_write_nodes.py [hunk 1]
```diff
@@ -61,4 +62,11 @@ def process(self, context):
                 plugin_data = plugin_data.copy()
                 plugin_data['WriteNode'] = node.name()
 
+                try:
+                    components = {node['fcompname'].getValue(): {}}
+                    instance.set_data('ftrackComponents', value=components)
```
Comment (tokejepsen) [inline pyblish_deadline/plugins/select_deadline_write_nodes.py]: This seems quite workflow specific, and it makes our render instances fail on conforming to ftrack. Any way of doing this outside, with a different plugin?

Comment (mkolar) [inline pyblish_deadline/plugins/select_deadline_write_nodes.py]: Are you not using the component name attribute created by ftrack?

Comment (tokejepsen) [inline pyblish_deadline/plugins/select_deadline_write_nodes.py]: No, we are just submitting to Deadline and letting the event plugin handle that logic.

**Comment by mkolar** [comment]
I was trying to stay with ftrack workflow as much as possible. I might just keep this in a production branch in my fork then. When we have multiple outputs from nuke file this is essential

**Comment by tokejepsen** [comment]
Could you add to the instances in a workflow plugin?
That would keep your deadline extension in sync with original repo.

**Comment by mkolar** [comment]
> Could you add to the instances in a workflow plugin?
That would keep your deadline extension in sync with original repo

should be able to. Will have a look at it

**Comment by mottosso** [comment]
Is this ready to be merged?

**Comment by tokejepsen** [comment]
No, I'll send a new pull request soon.

**Pull request closed without merging**